src/Serialization/parallel.jl

```diff
@@ -0,0 +1,266 @@
+  end
+end
+
+# Method for end of recursion.
+# TODO: there is probably a way to stop the code from getting here that will speed up the code
+function put_type_params(channel::RemoteChannel, ::TypeParams{T, Nothing}) where T
+  return
+end
+
+function put_type_params(channel::RemoteChannel,
+                         ::TypeParams{T, Tuple{Vararg{Pair{Symbol, Nothing}}}}) where T
+  return
+end
+
+# Recursive call. Send all subsequent parents on which this object is 
+# based and finally the object itself, if applicable. 
+function put_type_params(channel::RemoteChannel, tp::TypeParams)
+  # only  types that use ids need to be sent to the other processes
+  put_type_params(channel, params(tp))
+end
+
+function put_type_params(channel::RemoteChannel, tps::Tuple{Vararg{Pair}})
+  for tp in tps
+    put_type_params(channel, tp.second)
+  end
+end
+
+function put_type_params(channel::RemoteChannel, obj::T) where T
+  # only  types that use ids need to be sent to the other processes
+  if serialize_with_id(T)
+    put_type_params(channel, type_params(obj))
+    put!(channel, obj)
+  else
+    put_type_params(channel, type_params(obj))
+  end
+end
+
+#function put_type_params(channel::RemoteChannel, v::Vector)
+#  for tp in v
+#    put_type_params(channel, params(tp))
+#  end
+#end
+
+### Generic methods to (de-)serialize a concrete task. 
+function save_object(s::SerializerState, obj::T) where T <: ParallelTask
+  save_data_dict(s) do
+    for n in fieldnames(T)
+      n == :__attrs && continue
+      save_object(s, getfield(obj, n), Symbol(n))
+    end
+  end
+end
+
+function load_object(s::DeserializerState, ::Type{T}, params::Dict) where T <: ParallelTask
+  fields = []
+  for n in fieldnames(T)
+    n == :__attrs && continue
+    push!(fields, load_object(s, fieldtype(T, n), params[n], Symbol(n)))
+  end
+  return T(fields...)
+end
+
+# The method of `_compute` for the concrete task specifies what to do 
+# on the respective worker. The data will be extracted from the task's 
+# fields. The return value must be of the form `(success::Bool, result)` 
+# where `success` is to indicate whether we have an affirmative result 
+# in any reasonable sense. It is used to decide whether `all` or `any` 
+# of a given list of tasks to be done in parallel is achieved. 
+#
+# The second value `result` can be any reasonable result of the computation 
+# which is to be returned to the main node. Note that you must not create 
+# new parents on the worker which are required for the contents of `return`, 
+# i.e. they need to use the parent-like objects sent from the main node. 
+function _compute(::T) where T <: ParallelTask
+  error("please implement the function _compute for the type $T")
+end
+
+########################################################################
+# Generic implementations for deployment of tasks
+########################################################################
+@doc raw"""
+    parallel_all(
```

Comment:
```suggestion
    parallel_all(
```
Is there a docs (`.md`) file where these docstrings can be included?

Reply:
added